devdoc/interlocked_linux_requirements.md

```diff
@@ -0,0 +1,345 @@
+
+**SRS_INTERLOCKED_LINUX_43_006: [** `interlocked_and_16` shall return the initial value of `*destination`. **]**
+ 
+ ## interlocked_and_64
+
+```c
+MOCKABLE_FUNCTION(, int64_t, interlocked_and_64, volatile int64_t*, destination, int64_t, value);
+```
+**SRS_INTERLOCKED_LINUX_43_007: [** `interlocked_and_64` shall call `atomic_fetch_and` from `stdatomic.h`. **]**
+
+**SRS_INTERLOCKED_LINUX_43_008: [** `interlocked_and_64` shall return the initial value of `*destination`. **]**
+
+## interlocked_and_8
+
+```c
+MOCKABLE_FUNCTION(, int8_t, interlocked_and_8, volatile int8_t*, destination, int8_t, value);
+```
+**SRS_INTERLOCKED_LINUX_43_009: [** `interlocked_and_8` shall call `atomic_fetch_and` from `stdatomic.h`. **]**
+
+**SRS_INTERLOCKED_LINUX_43_010: [** `interlocked_and_8` shall return the initial value of `*destination`. **]**
+
+## interlocked_compare_exchange
+
+```c
+MOCKABLE_FUNCTION(, int32_t, interlocked_compare_exchange, volatile int32_t*, destination, int32_t, exchange, int32_t, comperand);
+```
+**SRS_INTERLOCKED_LINUX_43_011: [** `interlocked_compare_exchange` shall call `atomic_compare_exchange_strong` from `stdatomic.h`. **]**
+
+**SRS_INTERLOCKED_LINUX_43_012: [** `interlocked_compare_exchange` shall return the initial value of `*destination`. **]**
+
+## interlocked_compare_exchange_16
+
+```c
+MOCKABLE_FUNCTION(, int16_t, interlocked_compare_exchange_16, volatile int16_t*, destination, int16_t, exchange, int16_t, comperand);
+```
+**SRS_INTERLOCKED_LINUX_43_015: [** `interlocked_compare_exchange_16` shall call `atomic_compare_exchange_strong` from `stdatomic.h`. **]**
+
+**SRS_INTERLOCKED_LINUX_43_016: [** `interlocked_compare_exchange_16` shall return the initial value of `*destination`. **]**
+
+## interlocked_compare_exchange_64
+
+```c
+MOCKABLE_FUNCTION(, int64_t, interlocked_compare_exchange_64, volatile int64_t*, destination, int64_t, exchange, int64_t, comperand);
+
+```
+**SRS_INTERLOCKED_LINUX_43_017: [** `interlocked_compare_exchange_64` shall call `atomic_compare_exchange_strong` from `stdatomic.h`. **]**
+
+**SRS_INTERLOCKED_LINUX_43_018: [** `interlocked_compare_exchange_64` shall return the initial value of `*destination`. **]**
+
+## interlocked_compare_exchange_pointer
+
+```c
+MOCKABLE_FUNCTION(, void*, interlocked_compare_exchange_pointer, void* volatile*, destination, void*, exchange, void*, comperand);
+
+```
+**SRS_INTERLOCKED_LINUX_43_019: [** `interlocked_compare_exchange_pointer` shall call `atomic_compare_exchange_strong` from `stdatomic.h`. **]**
+
+**SRS_INTERLOCKED_LINUX_43_020: [** `interlocked_compare_exchange_pointer` shall return the initial value of `*destination`. **]**
+
+## interlocked_decrement
+
+```c
+MOCKABLE_FUNCTION(, int32_t, interlocked_decrement, volatile int32_t*, addend);
+
+```
+**SRS_INTERLOCKED_LINUX_43_021: [** `interlocked_decrement` shall call `atomic_fetch_sub` from `stdatomic.h`. **]**

```

Comment:
>atomic_fetch_sub [](start = 72, length = 16)

Again, this should really say what arguments you are passing (because you should say "1 as `operand`", which is not stated currently in your spec)